onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -324,6 +330,114 @@ const std::vector<std::string>& GetAvailableProviders() {
   return available_providers;
 }
 
+#ifdef USE_CUDA
+
+/*
+ * Validate a string that is positive integer or zero.
+ *
+ * (-1234, 43.21, +43.21 ..... are not valid) 
+ */
+bool IsPositiveInteger(const std::string & s)
+{
+   if(s.empty() || ((!isdigit(s[0])) && (s[0] != '+'))) { 
+       return false;
+    }
+
+   char * p;
+   std::strtol(s.c_str(), &p, 10);
+
+   return (*p == 0);
```

Comment:
Just a suggestion, another way of doing this is to: check if (p != s.c_str()  && the returned value from std::strtol >=0), then remove the first lines of this function.   A major difference between these two is: whether you want to allow the input string having leading spaces or not.

Reply:
I think using (p != s.c_str() && the returned value from std::strtol >=0) could be wrong under some cases.

Here is a case:
The default value of cuda memory size limitation is std::numeric_limits<size_t>::max() = 2^64-1=18446744073709551615 under 64 bits architecutre. 
However, the return type of std::strtol() is long (-(2^31) to (2^31)-1).
Therefore, when calling std::strtol("18446744073709551615", &p, 10) will cause range error and the return value of is negative.

So, I think I will keep the original version.

Reply:
Then you shouldn't use strtol in the first place
A simple rule: Don't use long type in our code, because it is 32 bits on Windows and 64 bits on Linux if the machine CPU is 64 bits. It brings in unnecessary complexity for dealing with the platform difference.

Reply:
Thanks for reminding, that's true long type has different implementations between Windows and Linux.
Therefore, I've used strtoll() instead, which returns unsigned long long type.
And both 64 bit Windows and 64 bit Linux CPU use 64 bits to implement unsigned long long type.